project/GenColCompatHelper.scala

```diff
@@ -67,6 +67,18 @@ object GenColCompatHelper {
           |  def newBuilder[A, C](f: Factory[A, C]): scala.collection.mutable.Builder[A, C] = f.newBuilder
           |
           |  type StringOps = scala.collection.StringOps
+          |
+          |  class InsertionOrderSet[A](elements: List[A]) extends scala.collection.immutable.Set[A] {
+          |    private val underlying = scala.collection.mutable.LinkedHashSet(elements: _*)
+          |    def contains(elem: A): Boolean = underlying.contains(elem)
+          |    def iterator: Iterator[A] = underlying.iterator
+          |    def excl(elem: A): scala.collection.immutable.Set[A] = new InsertionOrderSet(elements.filter(_ != elem))
+          |    def incl(elem: A): scala.collection.immutable.Set[A] = 
+          |      if (underlying.contains(elem)) 
+          |        new InsertionOrderSet(elements) 
+          |      else 
+          |        new InsertionOrderSet(if (underlying.contains(elem)) elements else elements :+ elem)
```

Comment:
this `if` was already checked and we know element is not there, we could just do `new InsertionOrderSet(elements :+ elem)`